spec/models/index_spec.rb

```diff
@@ -2,61 +2,37 @@
 
 RSpec.describe Index do
   describe "destroy behavior" do
-    let(:index) { create(:index) }
-    let(:section) { create(:index_section, index: index) }
-    let(:post) { create(:post) }
-    let(:sectionpost) { create(:index_post, index_section: section, index: index, post: post) }
+    let!(:index) { create(:index) }
+    let!(:section) { create(:index_section, index: index) }
+    let!(:post) { create(:post) }
+    let!(:sectionpost) { create(:index_post, index_section: section, index: index, post: post) }
 
     before(:each) do
-      post2 = create(:post)
-      sectionlesspost = create(:index_post, index: index, post: post2)
-      sectionpost # touch to create
-
-      expect(Index.count).to eq(1)
-      expect(IndexPost.count).to eq(2)
-      expect(IndexSection.count).to eq(1)
-      expect(Post.count).to eq(2)
+      create(:index_post, index: index)
     end
 
     it "should destroy indexposts and indexsections when deleting an index" do
-      index.destroy
-
-      expect(Index.count).to eq(0)
-      expect(IndexPost.count).to eq(0)
-      expect(IndexSection.count).to eq(0)
-      expect(Post.count).to eq(2)
+      expect {
+        index.destroy!
+      }.to change{Index.count}.by(-1).and change{IndexPost.count}.by(-2).and change{IndexSection}.by(-1).and not_change{Post.count}
```

Comment:
Is there a way to make this more legible on multiple lines?

Reply:
Having researched instead of just going off what's already in our code, it looks like yes!